### PR TITLE
send scheduled notifications for user

### PR DESCRIPTION
**Pull request by Proladge**

### CODE REVIEW

File: server/src/bots/telegram/services/storage.ts [hunk 1]
```diff
@@ -8,24 +8,44 @@ import {
...
+export const getTelegramActiveSubscriptions: Function = getActiveSubscriptions(
+    TELEGRAM_PREFIX
+);
+export const getTelegramUserSubscriptions: Function = getUserSubscription(
+    TELEGRAM_PREFIX
+);
+export const getTelegramActiveUserSubscriptions: Function = getActiveUserSubscription(
+    TELEGRAM_PREFIX
+);
+export const getTelegramUser = getUser(TELEGRAM_PREFIX);
```
Comment (danbilokha) [inline server/src/bots/telegram/services/storage.ts:32]: why some functions have types and some don't ?

File: server/src/services/infrastructure/scheduler.ts [hunk 2]
```diff
@@ -14,3 +17,22 @@ export const checkCovid19Updates = () => {
...
+            logger.error(
+                'sendReleaseNotificationToUsers failed when accessing users Db',
+                err,
+                LogCategory.Scheduler
+            );
+            return;
+        }
+
+        for (const usr of users) {
+            bot.sendMessage(usr.chatId, 'Hello DUDE, What\'s UP? ');
```
Comment (danbilokha) [inline server/src/services/infrastructure/scheduler.ts:35]: should we merge it? :D

File: server/src/bots/telegram/services/storage.ts (hunk 1 above)
Comment (Proladge) [reply server/src/bots/telegram/services/storage.ts:32]: I feel like this layer is simply decorator for injecting platoform in our storage.
So I'd say this layer don't need to know anyhing about the types which pure starage has already cooked.

We could enhance those who has types here.

File: server/src/services/infrastructure/scheduler.ts (hunk 2 above)
Comment (Proladge) [reply server/src/services/infrastructure/scheduler.ts]: Your question is super fair.
I've just made message more configurable

File: server/src/environments/environment.ts [hunk 3]
```diff
@@ -42,13 +48,19 @@ let envConfig = {
     },
     IsNgRokMode() {
         return this.IsDevelopment();
-    }
+    },
+    features: {
+        ScheduledNotification: {
+            enabled: process.env.EnableScheduledNotification || false,
+            cronExpr: process.env.ScheduledNotificationCronExpr,
```
Comment (danbilokha) [inline server/src/environments/environment.ts:55]: what is Expr?

File: server/src/services/infrastructure/scheduler.ts [hunk 4]
```diff
@@ -14,3 +22,52 @@ export const checkCovid19Updates = () => {
         );
     });
 };
+
+export const sendReleaseNotificationToUsers = async (bot: TelegramBot) => {
```
Comment (danbilokha) [inline server/src/services/infrastructure/scheduler.ts:26]: return type

File: server/src/services/infrastructure/scheduler.ts [hunk 5]
```diff
@@ -14,3 +22,52 @@ export const checkCovid19Updates = () => {
...
+            if (!message) {
+                logger.error(
+                    'getTelegramNotificationMessage() returns empty message',
+                    err,
+                    LogCategory.Scheduler
+                );
+            }
+
+            for (const usr of users) {
+                message.replace('#UserName#', getUserName(usr as any));
```
Comment (danbilokha) [inline server/src/services/infrastructure/scheduler.ts:68]: as any? 👎

File: server/src/services/infrastructure/scheduler.ts [hunk 6]
```diff
@@ -14,3 +22,52 @@ export const checkCovid19Updates = () => {
...
+                bot.sendMessage(usr.chatId, 'Hello DUDE, What\'s UP? ');
```
Comment (danbilokha) [inline server/src/services/infrastructure/scheduler.ts:69]: I still can see this message

File: server/src/bots/telegram/services/storage.ts (hunk 1 above)
Comment (danbilokha) [reply server/src/bots/telegram/services/storage.ts:33]: Actually, I feel that it should and The storage should not know anything about them as **storage** retrieves only data from, from different parts of it. On top of that, these different parts may have different data structure (preferably don't, but maybe some of them need for some kind of structural things), thus, data types have to be here, and in storage, I deem that should be generic. What do you think?

File: server/src/environments/environment.ts (hunk 3 above)
Comment (Proladge) [reply server/src/environments/environment.ts:55]: Expression
was not it obvious?

File: server/src/services/infrastructure/scheduler.ts (hunk 6 above)
Comment (Proladge) [reply server/src/services/infrastructure/scheduler.ts]: fixed

File: server/src/services/infrastructure/scheduler.ts (hunk 5 above)
Comment (Proladge) [reply server/src/services/infrastructure/scheduler.ts]: fixed

File: server/src/services/infrastructure/scheduler.ts (hunk 4 above)
Comment (Proladge) [reply server/src/services/infrastructure/scheduler.ts]: fixed

File: server/src/services/infrastructure/scheduler.ts [hunk 7]
```diff
@@ -14,3 +22,58 @@ export const checkCovid19Updates = () => {
...
 };
+
+export const sendReleaseNotificationToUsers = async (
+    bot: TelegramBot
+): Promise<void> => {
+    // At 08:00 PM, every day
+    schedule.scheduleJob(
+        environments.features.ScheduledNotification.cronExpr || '0 0 20 * * *',
+        async () => {
+            if (!environments.features.ScheduledNotification.enabled) {
```
Comment (danbilokha) [inline server/src/services/infrastructure/scheduler.ts:33]: move further for 3 lines (before scheduling) `schedule.scheduleJob(`

File: server/src/services/infrastructure/scheduler.ts [hunk 8]
```diff
@@ -14,3 +22,58 @@ export const checkCovid19Updates = () => {
...
+
+            for (const usr of users) {
+                message.replace('#UserName#', getUserName(usr));
+                bot.sendMessage(usr.chatId, message);
+            }
+        }
+    );
+};
+
+function getUserName(user: User): string {
```
Comment (danbilokha) [inline server/src/services/infrastructure/scheduler.ts:77]: utils

**Pull request merged**